drivers/pn532/pn532.c

```diff
@@ -172,15 +170,13 @@ static int _write(const pn532_t *dev, char *buff, unsigned len)
     (void)buff;
     (void)len;
 
-    if (dev->mode == PN532_I2C) {
 #ifdef PN532_SUPPORT_I2C
+    if (dev->mode == PN532_I2C) {
         i2c_acquire(dev->conf->i2c);
         ret = i2c_write_bytes(dev->conf->i2c, PN532_I2C_ADDRESS, buff, len);
         i2c_release(dev->conf->i2c);
-#endif
     }
     else {
-#ifdef PN532_SUPPORT_SPI
```

Comment:
Note i2c vs spi here.

Reply:
because of this, the 2 blocks are built depending on these 2 bus flags.